test/support/test_complex.h

```diff
@@ -86,14 +86,27 @@ run_test()
 //         // ...
 //     }
 #define IF_DOUBLE_SUPPORT(x)                                                                          \
-    if constexpr (HasDoubleSupportInRuntime::value) { x; }
+    TestUtils::invoke_test_if<>(HasDoubleSupportInRuntime(), [](){ x; });
 
 // We should use this macros to avoid compile-time error in code with long double type in Kernel.
 #define IF_LONG_DOUBLE_SUPPORT(x)                                                                     \
-    if constexpr (HasLongDoubleSupportInCompiletime::value) { x; }
```

Comment:
I rejected is constexpr because in case of it's usage I have compile errors which was found @ValentinaKats
I think (my assumption) that the root cause of these errors the code under "if sonstexpr" compiled in any case amd we have compile error.

Reply:
> I think (my assumption) that the root cause of these errors is that the code under "if constexpr" compiled in any case

Do we know if it is a compiler bug or "works as designed"?

Reply:
Seems it's compiler bug. Sent reproducer to dpcpp compiler team.